extensions/wikia/ImageReview/ImageReviewSpecialController.class.php

```diff
@@ -29,85 +75,74 @@ protected function setGlobalDisplayVars() {
+		$this->response->addAsset('extensions/wikia/ImageReview/css/ImageReview.scss');
 	}
 
 	public function index() {
-		$this->setGlobalDisplayVars();
-
 		$this->action = $this->parseAction();
-		$this->order = $this->getOrderingMethod();
-		$this->ts = $this->request->getVal( 'ts' );
+		$this->order = $this->request->getInt( 'sort' );
+		$this->timestamp = $this->request->getVal( 'ts' );
 
 		$this->checkUserPermissions();
-		$this->checkRedirect();
 
-		$this->setCache();
-		$this->setAssets();
-
-		$this->helper = $this->getHelper();
-
-		if( $this->wg->request->wasPosted() ) {
-			$this->handlePostAction();
+		if ( $this->checkHandleStats() ) {
+			// Return true to stop executing index but make sure to render the template
+			return true;
 		}
 
-		$user_key = $this->helper->getUserTsKey();
-		$newestTs = $this->wg->Memc->get( $user_key );
-
-		if ( $this->ts > $newestTs ) {
-			WikiaLogger::instance()->info( 'ImageReviewLog', [
-				'method' => __METHOD__,
-				'message' => "I've got the newest ts ({$this->ts}), I won't refetch the images",
-			]);
-			$this->imageList = array();
-			$this->wg->memc->set( $user_key, $this->ts, 3600 /* 1h */ );
-		} else {
-			$this->imageList = $this->helper->refetchImageListByTimestamp( $this->ts );
+		if ( $this->wg->request->wasPosted() ) {
+			$this->handlePostAction();
 		}
 
-		/* SUS-541 / Mix <mix@wikia.com> / scope: the following if block */
-		if ( count( $this->imageList ) < ImageReviewHelper::LIMIT_IMAGES ) {
-			$this->logImageListCompleteness( 'warning' );
+		if ( $this->checkRedirect() ) {
+			// Return false to stop any template rendering
+			return false;
 		}
 
-		if ( count( $this->imageList ) == 0 ) {
-			$do = array( 
-				self::ACTION_QUESTIONABLE	=> ImageReviewStatuses::STATE_QUESTIONABLE,
-				self::ACTION_REJECTED		=> ImageReviewStatuses::STATE_REJECTED,
-				'default'					=> ImageReviewStatuses::STATE_UNREVIEWED
-			);
-
-			if ( isset( $do[ $this->action ] ) ) {
-				$this->imageList = $this->helper->getImageList( $this->ts, $do[ $this->action ], $this->order );
-				$this->imageCount = $this->helper->getImageCount();
-			} else {
-				$this->imageList = $this->helper->getImageList( $this->ts, $do[ 'default' ], $this->order );
-				$this->imageCount = $this->helper->getImageCount( ImageReviewSpecialController::ACTION_UNREVIEWED, count( $this->imageList ) );
-			}
-		} else {
-			$this->imageCount = $this->helper->getImageCount();
-		}
+		$this->imageList = $this->getImageList();
+		$this->imageCount = $this->getImageCounts();
 
 		$this->setVariables();
 
-		$this->imageCount = array_map(function($number) {
-			return $this->wg->Lang->formatNum($number);
-		}, $this->imageCount);
-
 		/* SUS-541 / Mix <mix@wikia.com> / scope: the following if block */
-		$severity = count( $this->imageList ) < ImageReviewHelper::LIMIT_IMAGES ? 'error' : 'success';
+		$severity = count( $this->imageList ) < ImageListGetter::LIMIT_IMAGES && $this->action == self::ACTION_UNREVIEWED
```

Comment:
I added the following condition for this logging: `$this->action == self::ACTION_UNREVIEWED`. My reasoning is this logging (as I understand it) is meant to track cases where the Image Review Tool didn't return enough images to the user. This is mostly a problem encountered when fetching new images to review, however this logging was also being triggered when users worked on the "rejected" and "questionable" pages. Since the the number of rejected and questionable images are far lower, it's expected that those queues will be cleared out and the number of images returned is less than `ImageListGetter::LIMIT_IMAGES`.